extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -148,6 +161,52 @@ void HTTPClient::setDisableHostVerification() {
   curl_easy_setopt(http_session_, CURLOPT_SSL_VERIFYHOST, 0L);
 }
 
+bool HTTPClient::setSpecificSSLVersion(SSLVersion specific_version) {
+#if CURL_AT_LEAST_VERSION(7, 54, 0)
+  CURLcode ret = CURLE_UNKNOWN_OPTION;
+  switch (specific_version) {
+    case SSLVersion::SSLv2:
+      ret = curl_easy_setopt(http_session_, CURLOPT_SSLVERSION, CURL_SSLVERSION_SSLv2);
```

Comment:
This function (setting the only SSL/TLS version we are willing to support) is really only for testing the ListenHTTP processor's "SSL Minimum Version" function. I would be more than happy not to support SSLv2 and SSLv3 at all, and even the test does not need that, because I can do it with different TLS versions.

Reply:
Correct. NiFi currently only supports TLS v1.2 for incoming connections and has backward compatibility to use TLS v1.0 and v1.1 for outgoing connections to support legacy systems. We recommend taking this approach.

This page notes the changes and when they occurred:
https://cwiki.apache.org/confluence/display/NIFI/Migration+Guidance

Reply:
I am changing this to exclude SSLv2 and SSLv3, so only outgoing connections with TLSv1.0, TLSv1.1 and TLSv1.2 will be supported.
However, ListenHTTP's SSLMinimumVersion property (and the underlying implementation) supports every version from SSLv2 to TLSv1.2. While I completely agree with the approach of only allowing TLSv1.2 for incoming connections, this would mean an API change and a deprecation, which I am quite sure can not be made in 0.6.1 and not sure how it can be made in 0.7.0.
@phrocker What are your thoughts on this?

Reply:
@alopresto, what are your opinions?

Reply:
@alopresto @thenatog Since then we talked about this with @phrocker and agreed that the full modifications recommended could be part of 0.7.0. I have already created a follow-up ticket for that: https://issues.apache.org/jira/browse/MINIFICPP-1005